tests/testament/categories.nim

```diff
@@ -63,6 +64,52 @@ proc compileRodFiles(r: var TResults, cat: Category, options: string) =
   test "gtkex1", true
   test "gtkex2"
 
+# --------------------- flags tests -------------------------------------------
+
+proc flagTests(r: var TResults, cat: Category, options: string) =
+  template test(filename: untyped, clearCacheFirst=false) =
+    if clearCacheFirst: delNimCache(filename, options)
+    testSpec r, makeTest(flagsDir / filename, options, cat)
+
+  # --genscript
+  test "tgenscript", true
+  test "tgenscript"
+
+  var build = ""
+  var run = ""
+  var cmd = "cd " & flagsDir / "nimcache" & " && "
+  var script = "compile_tgenscript"
+  when defined(windows):
+    cmd = "cmd /c " & cmd & "$#"
+    script &= ".bat"
+
+  when defined(linux):
+    cmd = "sh -c \"" & cmd & "./$#\""
+    script &= ".sh"
+
+    setFilePermissions(flagsDir / "nimcache" / script,
+      {fpUserExec, fpUserWrite, fpUserRead, fpGroupExec, fpGroupRead,
+       fpOthersExec, fpOthersRead})
+
+  build = cmd % script
+  run = cmd % "tgenscript"
+
+  # Build
+  var (outp, errC) = execCmdEx(build)
+  r.total += 1
```

Comment:
There should be procs for this, don't mess manually with r.total and r.passed here.